c++/include/orc/Reader.hh

```diff
@@ -288,6 +288,17 @@ namespace orc {
     virtual uint64_t getCompressionSize() const = 0;
 
     /**
+     * Get ID of writer that generated the file.
+     * Current availiable Orc writers:
+     * 0 = ORC Java
+     * 1 = ORC C++
+     * 2 = Presto
+     * @param id out parameter for writer id
+     * @return true if writer id is availiable, false if otherwise
+     */
+    virtual bool getWriterId(uint32_t & id) const = 0;
```

Comment:
If it isn't there, it means ORC Java. Let's make an enum that matches this:

    enum WriterId { ORC_Java = 0, ORC_CPP = 1, ORC_Presto = 2, ORC_FUTURE = INT_MAX }

Any unknown code should return ORC_FUTURE.

   virtual WriterId getWriterId() const = 0;

Reply:
Actually, we do have the error stream, but we don't need to use it. I'd prefer not to use side effects in the parameters, because it is easy to misread in the code.

For the code and most users, the important part is which writer it is if known. Only diagnostic tools will care about the integer value of unknown ones. How about:

    enum WriterId { ORC_JAVA_WRITER, ORC_CPP_WRITER, PRESTO_WRITER, UNKNOWN_WRITER}

    class Reader {
    
       WriterId getWriterId();
       /**
         * For unknown writer ids, get the value.
         */
       int getUnknownWriterIdValue();

Reply:
ok, that sounds good. I will update it in my next change.